### DISCUSSION TITLE
blog/cs/2024-08-07-OSI-7-Layer/

### DISCUSSION THREAD

**Discussion by giscus** [mingdodev/mingdodev.github.io#20] [Announcements]
# blog/cs/2024-08-07-OSI-7-Layer/

인터넷에서 컴퓨터들이 서로 정보를 주고받을 때에는 각각의 상황에 따른 통신 규약이 필요하다. 이러한 통신 규약을 프로토콜이라고 하며, 이를 통해 네트워크 통신이 일어나는 과정을 OSI 7 계층 모델 또는 TCP/IP 4계층 모델로 설명할 수 있다. 컴퓨터들 간의 통신은 사람이 말을 뱉고, 말이 공기를 통해 전달되고, 다른 사람이 말을 듣는 대화의 과정 혹은 편지를 쓰고, 내용을 암호화한 뒤, 봉투에 넣어 우편을 보내는 과정 에 빗대어 이해할 수 있다. 이러한 하나하나의 동작이 계층을 이루게 되며, 이 계층들 간에는 상관 관계가 없다. 오로지 각 계층에서 산출된 결과만 전달할 뿐이다.

https://mingdodev.github.io/blog/cs/2024-08-07-OSI-7-Layer/

**Comment by mikio999** [comment]
네트워크 계층에서 IP 주소를 통해 경로를 탐색한다고 했는데, 라우팅 알고리즘의 종류와 그 작동 방식에 대해 설명해 주실 수 있나요?

**Reply by mingdodev** [reply]
관리자가 직접 경로를 지정하는 것이 아닌 경우, 네트워크의 변화에 따라 다음과 같은 라우팅 알고리즘으로 경로를 탐색할 수 있습니다!
- **Distance vector algorithm**
    - 인접 라우터들로부터 거리 정보를 받아 얻은 데이터로 최적의 경로를 계산하는 방식
    - 주기적으로 인접 라우터와 정보를 교환하여 라우팅 테이블을 업데이트
    - 벨만-포드 알고리즘 사용

- **Link-State Routing Algorithm**
    - 네트워크 전체에 대한 정보를 파악하여 최적의 경로를 계산하는 방식
    - 모든 라우터들이 서로의 연결 상태를 공유하여 전체 네트워크를 파악할 수 있음
    - 다익스트라 알고리즘 사용
